integrations/salesforce/oauth.go

```diff
@@ -88,6 +110,63 @@ func (h handler) saveConnection(ctx context.Context, vsid sdktypes.VarScopeID, t
 	for k, v := range extra {
 		vs = vs.Append(sdktypes.NewVar(sdktypes.NewSymbol(k)).SetValue(fmt.Sprintf("%v", v)))
 	}
+	vs = vs.Append(sdktypes.NewVar(orgIDVar).SetValue(orgID))
+	vs = vs.Append(sdktypes.NewVar(common.OAuthExpiryVar).SetValue(exp.Format(time.RFC3339)))
 
 	return h.vars.Set(ctx, vs.WithScopeID(vsid)...)
 }
+
+func (h handler) accessTokenExpiration(ctx context.Context, instanceURL, accessToken string, cid sdktypes.ConnectionID) (time.Time, error) {
+	vs, errStatus, err := common.ReadConnectionVars(ctx, h.vars, cid)
+	if errStatus.IsValid() || err != nil {
+		return time.Time{}, err
+	}
+
+	formData := url.Values{
+		"token":           {accessToken},
+		"token_type_hint": {"access_token"},
+		"client_id":       {vs.GetValueByString("private_client_id")},
+		"client_secret":   {vs.GetValueByString("private_client_secret")},
+	}
+	encodedFormData := formData.Encode()
+
+	u, err := url.JoinPath(instanceURL, "services", "oauth2", "introspect")
```

Comment:
In contrast to OS path, URLs can use only "/" as a separator, so no need to split elements in the path
```suggestion
	u, err := url.JoinPath(instanceURL, "services/oauth2/introspect")
```

Reply:
Done